src/script/calling/CallingRepository.js

```diff
@@ -960,38 +961,39 @@ export class CallingRepository {
       const isActiveCall = conversationEntity.id === this._selfClientOnACall();
       return isActiveCall
         ? this.leaveCall(conversationEntity.id, TERMINATION_REASON.SELF_USER)
-        : this.joinCall(conversationEntity.id, mediaType);
+        : this.joinCall(conversationEntity, mediaType);
     }
   }
 
   /**
    * Check whether conversation supports calling.
    *
    * @private
-   * @param {string} conversationId - ID of conversation to join call in
+   * @param {Conversation} conversationEntity - conversation to join call in
    * @param {z.media.MediaType} mediaType - Media type for this call
    * @param {CALL_STATE} callState - Current state of call
    * @returns {Promise} Resolves when conversation supports calling
    */
-  _checkCallingSupport(conversationId, mediaType, callState) {
-    return this.conversationRepository.get_conversation_by_id(conversationId).then(conversationEntity => {
+  _checkCallingSupport(conversationEntity, mediaType, callState) {
+    return new Promise((resolve, reject) => {
```

Comment:
I reduced the dependency to the `conversationRepo` here by injecting the conversationEntity directly into the method.
In the near future, the conversationRepo should be completely gone from the CallingRepo